addons/hr_homeworking/models/hr_homeworking.py

```diff
@@ -0,0 +1,293 @@
+            # self.unlink()
+            return True
+        else:
+            self.env.cr.execute(
+                'UPDATE hr_homeworking SET current=NULL WHERE id=%s',
+                (self.id,)
+            )
+            self.end = fields.Date().today() - timedelta(days=1)
+            self.child_removed_ids.filtered(lambda a : a.date >= fields.Date.today()).unlink()
+            return False
+
+
+    def _next_weekday_date_after_today(self, date):
+        return fields.Date().today() + timedelta(days= ((date.weekday() - fields.Date.today().weekday() +7)%7))
+
+    def _removed_worklocation_to_active_worklocation(self, work_location_id):
+        self.removed = False
+        self.parent_default_homeworking = False
+        self.work_location_id = work_location_id
+
+    def action_register_new_day_work_location(self):
+        print("coucou")
+
+    def _check_exceptional_work_location(self,vals):
+        date = vals.get("date") or self.date
+        work = vals.get("work_location_id") or self.work_location_id
+        employee_id = vals.get("employee_id") or self.employee_id.id
+        exception_work_location = self.env['hr.homeworking'].search([
+            ('date',"=", date),
+            ("employee_id.id", "=", employee_id),
+            ("weekly", "=", False),
+            ("id", "!=", self.id)
+        ])
+        return exception_work_location, date, work, employee_id
+
+
+    def add_exceptional_worklocation(self,vals):
+        exceptional, date, work, employee_id = self._check_exceptional_work_location(vals)
+        if exceptional:
+            exceptional.unlink()
+        self.env['hr.homeworking'].create({
+            'work_location_id' : work,
+            'employee_id' : employee_id,
+            'date' : date,
+        })
+
+        
+    def exceptional_worklocation_to_default_worklocation(self, vals, no_unlink=False):
+        if vals.get("date"):
+            default_date = string_to_datetime(vals.get("date")).date()
+            date = default_date
+        else:
+            default_date = self.date
+            date = default_date
+        default_work = self.env['hr.homeworking'].search([
+                ("employee_id.id", "=", self.employee_id.id),
+                ("weekday", "=", default_date.weekday()),
+                ("weekly", "=", True),
+                ("current", "=", True)
+            ])
+        removed_work = self.env['hr.homeworking'].search([
+                ("employee_id.id", "=", self.employee_id.id),
+                ("date", "=", date),
+                ("removed", "=", True),
+            ])
+        if removed_work:
+            removed_work.unlink()
+        if default_date < fields.Date().today():
+            default_date = self._next_weekday_date_after_today(date)
+        if default_work:
+            res = default_work._save_previous_default_worklocation()
+            if res:
+                default_work.child_removed_ids.unlink()
```

Comment:
Shouldn't be necessary by using ondelete cascace
```py
    parent_default_homeworking = fields.Many2one("hr.homeworking", help="removed records from default record", ondelete='cascade')
    child_removed_ids = fields.One2many("hr.homeworking",'parent_default_homeworking')
```

Reply:
ohhhhhhhhhhh thanks :o